distributed/tests/test_scheduler.py

```diff
@@ -1068,6 +1068,30 @@ def test_service_hosts_match_scheduler():
         yield s.close()
 
 
+@pytest.mark.skipif(not sys.platform.startswith('linux'),
+                    reason="Need 127.0.0.* to mean localhost")
+@gen_test(timeout=None)
+def test_service_custom_host():
+    pytest.importorskip('bokeh')
+    from distributed.bokeh.scheduler import BokehScheduler
+    services = {('bokeh', ('127.0.0.3', 0)): BokehScheduler}
+
+    s = Scheduler(services=services)
+    yield s.start('tcp://0.0.0.0')
+
+    sock = first(s.services['bokeh'].server._http._sockets.values())
+    assert sock.getsockname()[0] in ('::', '0.0.0.0')
```

Comment:
I'm curious, why wouldn't this be the hostname that we specified above?  (My understanding of networking here is poor)

Reply:
You're right, it should, and it's confirmed by the Travis failure.

It seems that when I run `py.test distributed --verbose` it's silently exiting during `distributed/deploy/tests/test_ssh.py::test_cluster` without indicating any sort of failure, and hadn't even reached my new test. So I was falsely assuming it was passing.